chimney/src/test/scala/io/scalaland/chimney/WixSpec.scala

```diff
@@ -110,7 +110,34 @@ object WixSpec extends TestSuite {
     }
 
     "support java enum" - {
+      "allow `withCoproductInstance` with java enum values" - {
+        implicit val t: Transformer[JavaColors.Colors, richcolors.RichColor] =
+          Transformer
+            .define[JavaColors.Colors, richcolors.RichColor]
+            .withCoproductInstance{_: JavaColors.Colors.Black.type => richcolors.JetBlack}
+            .withCoproductInstance{_: JavaColors.Colors.Red.type => richcolors.SalmonRed}
+            .withCoproductInstance{_: JavaColors.Colors.Green.type => richcolors.SeawaveGreen}
+            .withCoproductInstance{_: JavaColors.Colors.Blue.type => richcolors.SkyBlue}
+            .buildTransformer
+        t.transform(JavaColors.Colors.Black) ==> richcolors.JetBlack
```

Comment:
I would actually check for all values, e.g. black, red, green and blue. Maybe chimney picks only the first coproduct, who knows?

Reply:
There's a test later, that verifies the customization works correctly. This one is just to verify that the syntax is allowed
https://github.com/wix-incubator/chimney/blob/master/chimney/src/test/scala/io/scalaland/chimney/WixSpec.scala#L155
https://github.com/wix-incubator/chimney/blob/master/chimney/src/test/scala/io/scalaland/chimney/WixSpec.scala#L191

Reply:
But they do not check that multiple usages of coproduct instances properly work, e.g. resulting transformer doesn't map all values into one value.

Reply:
Added tests in a different PR (since this one is already merged)
https://github.com/wix-incubator/chimney/pull/22

Reply:
I'm sorry, I was not trying to argue, just explained why I didn't check the values. Intended to add tests. Bad comm. on my side.